presto-spi/src/main/java/io/prestosql/spi/type/VarcharType.java

```diff
@@ -134,6 +135,24 @@ public int compareTo(Block leftBlock, int leftPosition, Block rightBlock, int ri
         return leftBlock.compareTo(leftPosition, 0, leftLength, rightBlock, rightPosition, 0, rightLength);
     }
 
+    @Override
+    public Optional<Range> getRange()
+    {
+        if (length > 100) {
+            return Optional.empty();
+        }
+
+        int codePointSize = SliceUtf8.lengthOfCodePoint(Character.MAX_CODE_POINT);
+
+        Slice slice = Slices.allocate(codePointSize * length);
+        int position = 0;
+        for (int i = 0; i < length; i++) {
+            position = SliceUtf8.setCodePointAt(Character.MAX_CODE_POINT, slice, position);
```

Comment:
```suggestion
            position += SliceUtf8.setCodePointAt(Character.MAX_CODE_POINT, slice, position);
```
?

(please add a test)

Reply:
Good catch